examples/run_glue.py

```diff
@@ -202,5 +190,31 @@ def compute_metrics(p: EvalPrediction) -> Dict:
     return results
 
 
+def _mp_fn(rank, model_args, data_args, training_args):
+    run_glue(model_args, data_args, training_args)
+
+
+def main():
+    # See all possible arguments in src/transformers/training_args.py
+    # or by passing the --help flag to this script.
+    # We now keep distinct sets of args, for a cleaner separation of concerns.
+
+    parser = HfArgumentParser((ModelArguments, DataTrainingArguments, TrainingArguments))
+
+    if len(sys.argv) == 2 and sys.argv[1].endswith(".json"):
+        # If we pass only one argument to the script and it's the path to a json file,
+        # let's parse it to get our arguments.
+        model_args, data_args, training_args = parser.parse_json_file(json_file=os.path.abspath(sys.argv[1]))
+    else:
+        model_args, data_args, training_args = parser.parse_args_into_dataclasses()
+
+    if training_args.use_tpu:
+        import torch_xla.distributed.xla_multiprocessing as xmp
+
+        return xmp.spawn(_mp_fn, args=(model_args, data_args, training_args), nprocs=training_args.num_cores)
+    else:
+        return run_glue(model_args, data_args, training_args)
+
+
```

Comment:
This seems like something we would need to put in every script, so there's probably a way to cleanly factor this in the trainer.